src/tribler-core/tribler_core/components/ipv8.py

```diff
@@ -89,28 +89,40 @@ async def run(self):
             'asyncio', 'attestation', 'dht', 'identity', 'isolation', 'network', 'noblockdht', 'overlays'
         ])
 
+    def initialise_community_by_default(self, community):
+        community.bootstrappers.append(self.make_bootstrapper())
+
+        # Value of `target_peers` must not be equal to the value of `max_peers` for this community.
+        # This causes a deformed network topology and makes it harder for peers to connect to others.
+        # More information: https://github.com/Tribler/py-ipv8/issues/979#issuecomment-896643760
+        random_walk_max_peers = max(10, community.max_peers - 10)
+        self.ipv8.add_strategy(community, RandomWalk(community), random_walk_max_peers)
+
+    async def unload_community(self, community):
+        await self.ipv8.unload_overlay(community)
+
     def make_bootstrapper(self) -> DispersyBootstrapper:
         args = DISPERSY_BOOTSTRAPPER['init']
         if bootstrap_override := self.session.config.ipv8.bootstrap_override:
             address, port = bootstrap_override.split(':')
             args = {'ip_addresses': [(address, int(port))], 'dns_addresses': []}
         return DispersyBootstrapper(**args)
 
-    def init_peer_discovery_community(self):
+    def _init_peer_discovery_community(self):
         ipv8 = self.ipv8
         community = DiscoveryCommunity(self.peer, ipv8.endpoint, ipv8.network, max_peers=100)
+        community.bootstrappers.append(self.make_bootstrapper())
```

Comment:
**nit:** seems like you can replace this with `self.initialise_community_by_default(community)` and also shed the `ipv8.add_strategy(community, RandomWalk(community), 20)`.
Same for `_init_dht_discovery_community`. Loses 2 LOC overall.

Reply:
Agree. Is 20 target peers is necessary for the RandomWalk strategy for DiscoveryCommunity? 

Why I asking is because  if I replace it by `self.initialise_community_by_default(community)`, then `target_peers` become `max_peers-10 == 90`

Reply:
Short answer: yes, 20 is a good idea.

Long answer: The DiscoveryCommunity is a meta-community that all peers join to quickly synchronize the other Communities they are a part of. This also forms a secondary bootstrap mechanism (which is why you generally don't want to deviate from the default parameterization of your Community and use a custom `Network()`). Actively walking to some peers in the DiscoveryCommunity  will allow you to get a little bit of diversity in your connections. However, walking to too many peers here will just give you a lot of peers you do not share Communities with. 20 seems to be a good number (also considering the network topologies of the other Communities this Community shares its peer pool with). You could even set this to 0. However, 90 may produce too many open connections if overlay identifiers start diverging in the network.

Reply:
Got it. What if we will rewrite `initialise_community_by_default` this way:

```python
    def initialise_community_by_default(self, community, default_random_walk_max_peers=20):
        community.bootstrappers.append(self.make_bootstrapper())

        # Value of `target_peers` must not be equal to the value of `max_peers` for the community.
        # This causes a deformed network topology and makes it harder for peers to connect to others.
        # More information: https://github.com/Tribler/py-ipv8/issues/979#issuecomment-896643760
        #
        # Then:
        # random_walk_max_peers should be less than community.max_peers:
        random_walk_max_peers = min(default_random_walk_max_peers, community.max_peers - 10)

        # random_walk_max_peers should be greater than 0
        random_walk_max_peers = max(0, random_walk_max_peers)
        self.ipv8.add_strategy(community, RandomWalk(community), random_walk_max_peers)
```

Reply:
Seems good to me 👍

Reply:
Implemented